sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/ScalaReflection.scala

```diff
@@ -590,7 +591,11 @@ object ScalaReflection extends ScalaReflection {
               "cannot be used as field name\n" + walkedTypePath.mkString("\n"))
           }
 
-          val fieldValue = Invoke(inputObject, fieldName, dataTypeFor(fieldType))
+          // primitive take only non-null or struct takes non-null object guarded by isNull
```

Comment:
The result is correct. But it looks a bit weird because actually the returned Tuple can be null.

You set it as non-null because we can infer correct nullability for its fields by that.

This part is little tricky to me. We may need more clear comment to explain this if we want to do this. Can you rephrase the current comment?